src/interceptors/XMLHttpRequest/utils/createResponse.ts

```diff
@@ -6,7 +8,19 @@ export function createResponse(
   request: XMLHttpRequest,
   body: BodyInit | null
 ): Response {
-  return new Response(body, {
+  /**
+   * Handle XMLHttpRequest responses that must have null as the
+   * response body when represented using Fetch API Response.
+   * XMLHttpRequest response will always have an empty string
+   * as the "request.response" in those cases, resulting in an error
+   * when constructing a Response instance.
+   * @see https://github.com/mswjs/interceptors/issues/379
+   */
+  const responseBodyOrNull = statusCodesWithoutBody.includes(request.status)
```

Comment:
I moved this check to the `createResponse()` function so it always produces the correct response no matter what XHR instance it's given. This logic belongs here.

Reply:
@avivasyuta, also note that representing the status codes that must not have a body in an array is more practical in this case than an object. At its core, we want a list of things, so an array is the right choice here!

Reply:
Yes, at first I thought that it would be more productive to use an object, but for three elements this can be neglected.